erc20-load-test-tool/src/main/java/com/vmware/ethereum/service/SecureTokenFactory.java

```diff
@@ -52,13 +62,45 @@
   private final DatabaseConfig dbConfig;
   private final Web3j web3j;
   private final Credentials credentials;
+  private final String senderAddress;
 
   /** Get SecureToken contract either by loading or deploying. */
   @SneakyThrows(Exception.class)
   public SecurityToken getSecureToken() {
-    ContractGasProvider gasProvider =
-        new StaticGasProvider(
-            valueOf(tokenConfig.getGasPrice()), valueOf(tokenConfig.getGasLimit()));
+    BigInteger gasEstimate = valueOf(tokenConfig.getGasLimit());
```

Comment:
Are the values of gasEstimate and gasLimit not fixed in VMBC? If so, we can just define them as part of configuration rather than make calls?

Reply:
As discussed with @ramkri123 , the flow for gasFree mode will be - 
during deployment -> genesis.json - gasLimit = 7fff...
Apart from conventional gas config, we provide a gasFree mode flag, when set to True
It makes call to get the estimated gas limit of tx and gas price, and we use this value in static gas provider.
It will make sure, even if gasFree mode deployment is not configured runs won't fail.
@ramkri123 Please add  in case I left something